File: 203.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        if not head:
            return head
        while head and head.val == val:
            head = head.next
        curr = head
        ret_val = curr
        if not curr:
            return curr
        next = curr.next
        while curr and next:
            if next.val == val:
                curr.next = next.next
                next = next.next
            else:
                curr = curr.next
                next = next.next
        return ret_val

File: 203.cpp
#include <algorithm>
#include <deque>
#include <iostream>
#include <map>
#include <set>
#include <unordered_map>
#include <unordered_set>
#include <vector>
struct ListNode {
  int val;
  ListNode *next;
  ListNode() : val(0), next(nullptr) {}
  ListNode(int x) : val(x), next(nullptr) {}
  ListNode(int x, ListNode *next) : val(x), next(next) {}
};
using namespace std;
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
  ListNode *removeElements(ListNode *head, int val) {
    if (!head) {
      return head;
    }
    while (head && head->val == val) {
      head = head->next;
    }
    ListNode *ans = head;
    ListNode *curr = head;
    if (!curr) {
      return curr;
    }
    ListNode *next = curr->next;
    while (curr && next) {
      if (next->val == val) {
        curr->next = next->next;
        next = next->next;
      } else {
        curr = curr->next;
        next = next->next;
      }
    }
    return ans;
  }
};
int main() {}